python_modules/dagster/dagster/_core/definitions/dependency.py

```diff
@@ -586,6 +586,10 @@ def node_name(self) -> str:
     def is_dynamic(self) -> bool:
         return self.output_def.is_dynamic
 
+    @property
+    def output_name(self) -> str:
+        return self.output_def.name
```

Comment:
There is a corresponding `input_name` on `NodeInputHandle`, so I added this for symmetry.